### DISCUSSION TITLE
Getting the default `settings` table when using `nvim-lspconfig`

### DISCUSSION THREAD

**Discussion by devansh08** [neovim/neovim#26368] [Q&A]
I'm trying to setup `pylsp` using `nvim-lspconfig` and `mason-lspconfig`. If I go with the default configuration it works fine, with a bunch of features I do not want `pylsp` to handle. So I tried passing a custom `settings` table (by converting [this](https://github.com/python-lsp/python-lsp-server/blob/develop/CONFIGURATION.md) to the lua table format) to the `setup` function. But this pretty much breaks it for me. There are no errors but none of the features work either.

What I was trying to find is if there is a way to get the default `settings` object that is being used so I can debug to see if I'm messing something up in my custom config. There are quite a few features I want to disable and the settings table in docs is quite huge to manually try all combinations. So I'm hoping if I can get the base settings lua table used by default, I can play around with that and debug.

Any ideas on this ?

Note: Do let me know if this is not the right place for `nvim-lspconfig` discussions. The README there send me here.

**Comment by wookayin** [comment]
I think nvim-lspconfig is the right place to ask this question, but here's some answer for you:

```
require"lspconfig.server_configurations.pylsp"
```
would give you the default settings object defined in the nvim-lspconfig's server configuration. Any missing values will be resolved to the default values depending on the LSP server implementation.

If you'd like to get the "applied" config on the LSP client instance after (`lspconfig.setup {...}`),
```
vim.lsp.get_active_clients({ name = "pylsp" })[1].config  
```
